ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -509,7 +509,7 @@ private RecommendationWrapper getRecommendationWrapperFromAmbariHostGroups() {
                 newConfigurationMap.put(stringMapEntry.getKey(), stringMapEntry.getValue());
             } else {
                 if(stringMapEntry.getValue() != null) {
-                    newConfigurationMap.get(stringMapEntry.getKey()).putAll(stringMapEntry.getValue());
```

Comment:
Isn't it better if you also merge the the old `newConfigurationMap.get(stringMapEntry.getKey())` with `stringMapEntry.getValue()`